iothub/device/devdoc/API Design.md

```diff
@@ -1,302 +1,169 @@
-/// <param name="cancellationToken">A cancellation token.</param>
 /// <remarks>
-/// This will create a single telemetry message and will not combine multiple calls into one message. Use <seealso cref="SendTelemetryAsync(IDictionary{string, dynamic}, string, IConventionHandler, CancellationToken)"/>. Refer to the documentation for <see cref="IConventionHandler"/> if you want to use a custom serializer.
+/// Use the <see cref="TelemetryMessage(TelemetryCollection)"/> constructor to pass in the formatted telemetry payload and an optional
+/// <see cref="IPayloadConvention"/> that specifies your payload serialization and encoding rules.
 /// </remarks>
-public Task SendTelemetryAsync(string telemetryName, object telemetryValue, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+/// <param name="telemetryMessage">The telemetry message.</param>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+/// <returns></returns>
+public Task SendTelemetryAsync(TelemetryMessage telemetryMessage, CancellationToken cancellationToken = default)
+```
+#### All related types
 
-/// <summary>
-/// Sends a collection of telemetry.
-/// </summary>
-/// <param name="telemetryDictionary">A dictionary of dynamic objects </param>
-/// <param name="componentName">The component name this telemetry belongs to.</param>
-/// <param name="conventionHandler">A convention handler that defines the content encoding and serializer to use for the telemetry message.</param>
-/// <param name="cancellationToken">A cancellation token.</param>
-/// /// <remarks>
-/// This will either use the <see cref="DefaultConvention"/> to define the encoding and use the default Json serailzier. Refer to the documentation for <see cref="IConventionHandler"/> if you want to use a custom serializer.
-/// </remarks>
-public Task SendTelemetryAsync(IDictionary<string, object> telemetryDictionary, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+```csharp
+public class TelemetryMessage : Message {
+    public TelemetryMessage(TelemetryCollection telemetryCollection);
+}
 
-/// <summary>
-/// Send telemetry using the specified message.
-/// </summary>
-/// <remarks>
-/// Use this method when you need to define custom properties for the message.
-/// </remarks>
-/// <param name="telemetryMessage">The custom implemented telemetry message</param>
-/// <param name="componentName">The component name this telemetry belongs to.</param>
-/// <param name="cancellationToken">A cancellation token.</param>
-public Task SendTelemetryAsync(Message telemetryMessage, string componentName = default, CancellationToken cancellationToken = default);
+public class TelemetryCollection : PayloadCollection {
+    public TelemetryCollection(string componentName=null, IPayloadConvention convention=null);
```

Comment:
make Collection accessible so that you can initialize it beforehand (without having to call .Add multiple times)

Reply:
telemetry payload => should be updatable after initialization